src/getCurrentPosition/getCurrentPosition.desktop.tsx

```diff
@@ -0,0 +1,96 @@
+import {GetCurrentPosition, GeolocationErrorCode} from './getCurrentPosition.types';
+
+type GoogleAPIsGeoLocateResponse = {
+    location: {
+        lat: number;
+        lng: number;
+    };
+    accuracy: number;
+};
+
+type FetchConfigWithOptions = {
+    method: string;
+    signal?: AbortSignal;
+};
+
+const BASE_URL = 'https://www.googleapis.com/geolocation/v1/geolocate';
+
+// api request config
+const requestConfig: FetchConfigWithOptions = {
+    method: 'POST',
+};
+
+const getCurrentPosition: GetCurrentPosition = (
+    success,
+    error,
+    options, // we will ignore options.maximumAge and options.enableHighAccuracy since cant pass it to geolocate api directly
+) => {
+    // emulate the timeout param with an abort signal
+    if (options?.timeout) {
+        const abortController = new AbortController();
+        setTimeout(() => {
+            abortController.abort();
+        }, options.timeout);
+        requestConfig.signal = abortController.signal;
+    }
+
+    const tempAPIToken = ''; // we get this token from our backend
```

Comment:
Let's make `getCurrentPosition` take a parameter for `apiToken`.

We can add this to the `options` object (the third parameter we take in `getCurrentPosition`). We can make this property optional so that we don't have to pass a token when using this function in other platforms